api/plugins/action/deploy_target_config.py

```diff
@@ -24,44 +17,67 @@ def run(self, tmp=None, task_vars=None):
                     project["deployTargetConfigs"],
-                    self._task.args.get('configs', []),
+                    configs,
                 )
                 result['changed'] = False
                 if len(changes) > 0:
                     for config in changes:
-                        if replace:
-                            DeployTargetConfig(self.client).delete(project['id'], changes['_existing_id'])
-
-                        if DeployTargetConfig(self.client).add(
-                                int(project['id']), 
-                                config['branches'],
-                                int(config['deployTarget']),
-                                config['pullrequests'],
-                                int(config['weight']) if 'weight' in config.keys() else 0
-                            ):
-                            result['result'].append(config)
+                        if replace and config.get('_existing_id'):
+                            self._display.vvvv(f"deleting config {config}")
+                            DeployTargetConfig(self.client).delete(
+                                project['id'], config['_existing_id'])
+
+                        self._display.vvvv(f"adding config {config}")
+                        addResult = DeployTargetConfig(self.client).add(
+                            int(project['id']),
+                            config['branches'],
+                            int(config['deployTarget']),
+                            config['pullrequests'],
+                            int(config['weight']) if 'weight' in config.keys() else 0
+                        )
+                        if addResult:
+                            config['id'] = addResult['id']
+                        else:
+                            config['failed'] = True
+                        result['result'].append(config)
                     result['changed'] = True
             elif state == "absent":
                 result['changed'] = False
                 for c in project["deployTargetConfigs"]:
-                    if DeployTargetConfig(self.client).delete(project['id'], c['id']):
+                    delete_desired = False
+                    for given_config in configs:
+                        if given_config['branches'] != c['branches']:
+                            continue
+                        if given_config['pullrequests'] != c['pullrequests']:
+                            continue
+                        if str(given_config['deployTarget']) != str(c['deployTarget']['id']):
+                            continue
+                        if str(given_config['weight']) != str(c['weight']):
+                            continue
+                        delete_desired = True
+                        self._display.vvvv(f"deleting config {c}")
+                        break
+                    if delete_desired and DeployTargetConfig(self.client).delete(project['id'], c['id']):
```

Comment:
Delete was causing all configs to be deleted from a project even if only one config was specified as `absent`.